### DISCUSSION TITLE
Builtin changes for v1?

### DISCUSSION THREAD

**Discussion by christophwille** [open-policy-agent/community#645] [OPA and Rego] [answered]
I have unit tests for various builtin scenarios, and those worked nicely up until going from 0.68 to v1. The other stuff (if) was easy to figure out (and guess from the release notes), however, with the builtins the error message doesn't help me much.

https://github.com/christophwille/dotnet-opa-wasm/pull/57#issuecomment-2558374000

On building the wasm I get `error: load error: 1 error occurred during loading: rego_parse_error: illegal capabilities: rego_v1 feature required for parsing v1 Rego`

Can someone please point me in the right direction of how-to fix-up .68 to v1 builtin regos?

**Comment by anderseknert** [comment, marked as answer]
Hi there!

If you've changed your policies to follow 1.0 conventions (removed `import rego.v1` and such), you'll need to get a capabilities.json file which contains the `rego_v1` capability, or else the parser is going to work as it did before 1.0, and won't recognize that as valid. You can find capabilities files, including v1.0.0 [here](https://github.com/open-policy-agent/opa/tree/main/capabilities).

Alternatively, if you want to keep using the old syntax for a while, you should be able to run the `opa build` command with the `--v0-compatible` flag, and that should hopefully work with the capabilities file that you have (but would require that you keep `rego.v1` imports).

Let me know how it goes!

**Reply by christophwille** [reply]
Ah, thanks for pointing me in the right direction - turns out my [concat-capabilities.ps1](https://github.com/christophwille/dotnet-opa-wasm/blob/master/sample-policies/concat-capabilities.ps1) isn't doing the right thing - it doesn't account for features, and it wasn't in the resulting unit testing capabilities file.
